### PR TITLE
Adds immolator laser gun

### PR DESCRIPTION
**Pull request by Regen1**
Adds an immolator laser gun, which has 8 shots and ignites the target.
image

Bigger image
biggerimage

:cl: 
rscadd: Adds the Immolator laser gun, a modified laser gun with 8 shots that will ignite mobs, can be made in R&D
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/energy/laser.dm [hunk 1]
```diff
@@ -98,6 +98,14 @@ obj/item/weapon/gun/energy/laser/retro
...
 	charge_cost = 500
 
+/obj/item/weapon/gun/energy/immolator
+	name = "Immolator laser gun"
+	desc = "A modified laser gun, shooting highly concetrated beams with higher intensity that ignites the target, for the cost of draining more power per shot"
+	icon_state = "immolator"
+	fire_sound = 'sound/weapons/laser3.ogg'
+	projectile_type = "/obj/item/projectile/beam/immolator"
+	charge_cost = 1400
+
```
Comment (Alex6511) [inline code/modules/projectiles/guns/energy/laser.dm]: Extra line here should be removed to preserve formatting.

**Comment by CupcakesNHacksaws** [comment]
I like it. I want to say add it to cargo since they get so little love compared to science, but at the same time it reminds me of the plasma research level in science.

**Comment by Spacemanspark** [comment]
:+1:

File: code/modules/projectiles/projectile/beams.dm [hunk 2]
```diff
@@ -38,6 +38,17 @@
 	weaken = 5
 	stutter = 5
 
+
+/obj/item/projectile/beam/immolator
+	name = "immolation beam"
+
+/obj/item/projectile/beam/immolator/on_hit(var/atom/target, var/blocked = 0)
+	. = ..()
```
Comment (Fox-McCloud) [inline code/modules/projectiles/projectile/beams.dm]: this should work, but this is really strange when `..()` would be just fine.

File: code/modules/projectiles/guns/energy/laser.dm [hunk 3]
```diff
@@ -98,6 +98,14 @@ obj/item/weapon/gun/energy/laser/retro
 	projectile_type = "/obj/item/projectile/beam/xray"
 	charge_cost = 500
 
+/obj/item/weapon/gun/energy/immolator
+	name = "Immolator laser gun"
+	desc = "A modified laser gun, shooting highly concetrated beams with higher intensity that ignites the target, for the cost of draining more power per shot"
+	icon_state = "immolator"
```
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/energy/laser.dm]: this will be invisible, in hands.

File: code/modules/projectiles/guns/energy/laser.dm [hunk 4]
```diff
@@ -98,6 +98,14 @@ obj/item/weapon/gun/energy/laser/retro
 	projectile_type = "/obj/item/projectile/beam/xray"
 	charge_cost = 500
 
+/obj/item/weapon/gun/energy/immolator
+	name = "Immolator laser gun"
+	desc = "A modified laser gun, shooting highly concetrated beams with higher intensity that ignites the target, for the cost of draining more power per shot"
+	icon_state = "immolator"
+	fire_sound = 'sound/weapons/laser3.ogg'
+	projectile_type = "/obj/item/projectile/beam/immolator"
+	charge_cost = 1400
```
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/energy/laser.dm]: You can get 7.1428571 shots off. Whut.

Charge costs should always deplete a gun from its max capacity to 0, and not leave a partial charge left.

File: code/modules/projectiles/projectile/beams.dm (hunk 2 above)
Comment (RemieRichards) [inline code/modules/projectiles/projectile/beams.dm]: the base proc returns 0/1, some (but not all) overrides also return 0/1, seems logical to assume something is or WAS expecting a return value from this proc.

Comment (Regen1) [inline code/modules/projectiles/projectile/beams.dm]: It's TG copypaste dark magic something

**Comment by Finalsong** [comment]
classy

File: code/modules/projectiles/guns/energy/laser.dm (hunk 3 above)
Comment (Regen1) [inline code/modules/projectiles/guns/energy/laser.dm]: It will use the default gun sprite, just like the Xray. etc

**Comment by taukausanake** [comment]
This weapon looks cool, but does the beam do any damage aside from setting the target on fire? Can't the temperature gun do that already, granted it's emagged?

**Comment by Regen1** [comment]
@taukausanake I think so, but this will add a high tier weapon that allows for the same, and the temperature gun isn't really used for anything other than fighting slimes.

Also, it seems that the temperature gun will not make people actually burst into flames (as the emagged message says), so I guess that is a bug, but outside of the scope of the PR

File: code/modules/projectiles/guns/energy/laser.dm [hunk 5]
```diff
@@ -98,6 +98,15 @@ obj/item/weapon/gun/energy/laser/retro
...
 
+/obj/item/weapon/gun/energy/immolator
+	name = "Immolator laser gun"
+	desc = "A modified laser gun, shooting highly concetrated beams with higher intensity that ignites the target, for the cost of draining more power per shot"
+	icon_state = "immolator"
+	fire_sound = 'sound/weapons/laser3.ogg'
+	projectile_type = "/obj/item/projectile/beam/immolator"
+	origin_tech = "combat=4;materials=4;magnets=3;plasmatech=2"
+	charge_cost = 1250
+
```
Comment (Alex6511) [inline code/modules/projectiles/guns/energy/laser.dm]: Extra white space here

**Comment by Regen1** [comment]
Ready for merge? @Fox-McCloud @TheDZD

**Comment by taukausanake** [comment]
@Regen1 Outside this PR, yeah I guess temperature gun is fine just changing temps. Would this gun work wonders on killing xenos? I have a few reasons for wanting a fiery laser

**Comment by TheDZD** [comment]
:fire:

**Pull request merged**